### DISCUSSION TITLE
Arabic (and Persian) characters in [slug] is not supported

### DISCUSSION THREAD

**Discussion by siakar** [vercel/next.js#26758] [Help]
when I use Persian(or Arabic) characters in [slug] in  the following code for fetching the details of a single product  in next js, 
-----------------------------------------------------------------------------------------
export async function getServerSideProps({ query: { slug } }) {
  const { data } = await axios.get(`http://localhost:8000/api/product/${slug}`)
  return {
    props: {
      product: data,
    },
  }
}
---------------------------------------------------------------------------------------------
it gives the following Error(However, it works well with English characters):
TypeError: Request path contains unescaped characters

**Comment by jurijtokarski** [comment]
Did you try to encode characters? URL supports a limited amount of characters and you need to encode other ones. Check [URL encoding](https://en.wikipedia.org/wiki/Percent-encoding) for more information on which characters are supported.
